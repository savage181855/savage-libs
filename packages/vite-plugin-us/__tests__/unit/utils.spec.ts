import {
	camelCaseToHyphen,
	injectExternalCssLink,
	unionRegex,
	padEndWithSpace
} from 'vite-plugin-us/utils/utils'

describe('utils', () => {
	test('camelCaseToHyphen', () => {
		expect(camelCaseToHyphen('bigOne')).toBe('big-one')
	})

	test('injectExternalCssLink', () => {
		const res = `;(function(links2) {
    window.addEventListener("DOMContentLoaded", () => {
      links2.forEach((v) => {
        const link = document.createElement("link");
        link.rel = "stylesheet";
        link.href = v;
        document.head.appendChild(link);
      });
    });
  })(["xxxx","xxxx"]);`
		expect(injectExternalCssLink(['xxxx', 'xxxx'])).toBe(res)
	})

	test('unionRegex', () => {
		const regs = [/hello/, /fine/]
		expect(unionRegex(regs)).toEqual(/hello|fine/)
	})

	test('padEndWithSpace', () => {
		const s = 'a'
		expect(padEndWithSpace(s, 5)).toBe('a    ')
	})
})
