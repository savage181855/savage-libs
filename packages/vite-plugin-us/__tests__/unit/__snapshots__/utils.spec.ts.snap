// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`utils > generateJsDataUrlByCode 1`] = `"data:application/javascript,function%20fn()%20%7B%0A%20%20%20%20%20%20return%20%22foo%22%3B%0A%20%20%20%20%7D"`;

exports[`utils > minifyCode > should minify css code 1`] = `
".header_1f-hhx{justify-content:normal!important}.operation-area_HPE0LG{flex:8;justify-content:end}
"
`;

exports[`utils > minifyCode > should minify js code 1`] = `
"function fn(){return\\"foo\\"}
"
`;

exports[`utils > removeCommentFromCode 1`] = `
"		console.log('bar')
		"
`;

exports[`utils > transform > should transform css code 1`] = `
".header_1f-hhx{justify-content:normal!important}.operation-area_HPE0LG{flex:8;justify-content:end}
"
`;

exports[`utils > transform > should transform js code 1`] = `
"function fn(){return\\"foo\\"}
"
`;
